### PR TITLE
Vendor: add setup instructions and explain constraints

### PR DESCRIPTION
**Pull request by glaubinix**

### CODE REVIEW

File: docs/setup-vendor.md [hunk 1]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+to grant a customer access to all versions.
+
+### Limit package access by minimum stability
+
+By default, your customers will have access to all versions of a package. For VCS repositories this includes all branches.
+By Setting a minimum accessible stability you can prevent customers from having access to versions that are not ready for
+production use. Composer recognizes the following [stabilities](https://getcomposer.org/doc/articles/versions.md#stabilities)
+(in order of stability): dev, alpha, beta, RC, and stable. If you have a VCS repository with the following versions:
+`dev-main`, `dev-feature-1`, `1.0-a1`, `1.0-b1`, `1.0-rc1` and `1.0` then setting the minimum stability to `RC` will only
+allow versions `1.0-rc1` and `1.0`.
```
Comment (naderman) [inline docs/setup-vendor.md:71]:
```suggestion
allow the customer to install versions `1.0-rc1` and `1.0`.
```

File: docs/setup-vendor.md [hunk 2]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
 
 Customers can use your packages in either an existing or a new Composer project.
 
+## Setting up a customer in Private Packagist
+
+You can manage your customers and grant them access to packages on the vendor page in Private Packagist.
+To create your first customer click on "Create Customer" and fill out the form on the next page.
+
+The customer's short name will be used in the Composer repository URL and the web UI URL where your customer sees packages
+they have access to and can download archives. Please be aware that changing the URL will change the web UI URL and
```
Comment (naderman) [inline docs/setup-vendor.md:13]:
```suggestion
they have access to and can download archives. Please be aware that changing the short name will change the web UI URL and
```

File: docs/setup-vendor.md [hunk 3]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+the Composer repository URL and will therefore break existing Composer projects. The minimum accesible stability
```
Comment (naderman) [inline docs/setup-vendor.md:14]:
```suggestion
the Composer repository URL and will therefore break existing Composer projects refering to the current URL. The minimum accesible stability
```

File: docs/setup-vendor.md [hunk 4]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+setting allows you to limit which versions of a package a customer has access to. More about that in the
+["Limit package access by minimum stability"](#limit-package-access-by-minimum-stability) section. You can also
+enable package source URLs for a customer. By default, Private Packagist removes source URLs for all packages
+assigned to a customer, e.g. for a git repository the URL to clone the repository, and customers are only able to install
+packages as archives. If your customer has access to your repositories and you want them to be able to install
+packages from source e.g. via git clone then you can enable the source code URL delivery to customers through Composer.
```
Comment (naderman) [inline docs/setup-vendor.md:20]: Can you refactor this long paragraph into a list of the form options with an explanation each?

File: docs/setup-vendor.md [hunk 5]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+
+Customers can also be created via our API using our [API client](https://github.com/packagist/private-packagist-api-client) with the following code snippet:
```
Comment (naderman) [inline docs/setup-vendor.md:22]:
```suggestion
You can also create customers using our API through the [API client](https://github.com/packagist/private-packagist-api-client) with the following code snippet:
```

File: docs/setup-vendor.md [hunk 6]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+
+```
+<?php
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+$client = new \PrivatePackagist\ApiClient\Client();
+$client->authenticate('api-token', 'api-secret');
+$customer = $client->customers()->create('Customer', false, 'customer', 'beta');
```
Comment (naderman) [inline docs/setup-vendor.md:31]: maybe at least leave comments for what each value is, so it's clear what e.g. the "false" means here?

File: docs/setup-vendor.md [hunk 7]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+```
+
+## Giving a customer access to a package 
+
+The packages section of a customer allows you to manage which packages and which versions of a package a customer has access to.
```
Comment (naderman) [inline docs/setup-vendor.md:36]:
```suggestion
The packages section of a customer allows you to manage which packages and which versions of these packages a customer has access to.
```

File: docs/setup-vendor.md [hunk 8]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+```
+
+Private Packagist provides different ways on how you can limit which versions your customers have access to. If multiple
+limits are set then only versions that match all the constraints will be available to the customer. You can also choose
+to grant a customer access to all versions.
+
+### Limit package access by minimum stability
+
+By default, your customers will have access to all versions of a package. For VCS repositories this includes all branches.
+By Setting a minimum accessible stability you can prevent customers from having access to versions that are not ready for
```
Comment (naderman) [inline docs/setup-vendor.md:67]:
```suggestion
By setting a minimum accessible stability you can prevent customers from having access to versions that are not ready for
```

File: docs/setup-vendor.md [hunk 9]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+to all packages the customer has access to unless otherwise specified. 
+
+### Limit package access by version number
+
+The version number limit allows you to use the same [version constraints](https://getcomposer.org/doc/articles/versions.md#writing-version-constraints)
+that you are already familiar with from your composer.json files.
+
+This is useful if your customers buy a package license for `acme/library` version 1 and receive maintenance updates for
+that version but will have to purchase a new license for a future version 2 of the package. You can set this up with the
+constraint `^1.0`.
```
Comment (naderman) [inline docs/setup-vendor.md:83]:
```suggestion
that version but will have to purchase a new license for a future version 2 of the package. You can achieve this by
restricting them to releases of the major version 1 with the constraint `^1.0`.
```

File: docs/setup-vendor.md [hunk 10]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+
+### Limit package access by publication date
+
+If enabled then the customer will only have receive access to versions that were released before the date set. However,
```
Comment (naderman) [inline docs/setup-vendor.md:87]:
```suggestion
If enabled then the customer will only receive access to versions that were or will be released before the date set. However,
```

File: docs/setup-vendor.md [hunk 11]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
...
+they do keep having access to those versions even after that date.
```
Comment (naderman) [inline docs/setup-vendor.md:88]:
```suggestion
they keep access to those versions beyond the specified date. If you want to remove their access even to past versions, you need do delete the package from their list entirely.
```

**Review by naderman — CHANGES_REQUESTED** [review]

File: docs/setup-vendor.md [hunk 12]
```diff
@@ -4,6 +4,92 @@ Using Private Packagist for Vendors, your customers can install packages with Co
```
[the remark this answers is not in the document]
Comment (glaubinix) [reply docs/setup-vendor.md:4]: removed

File: docs/setup-vendor.md (hunk 4 above)
Comment (glaubinix) [reply docs/setup-vendor.md:20]: done

File: docs/setup-vendor.md (hunk 6 above)
Comment (glaubinix) [reply docs/setup-vendor.md]: added

**Comment by glaubinix** [comment]
Addressed all comments

File: docs/setup-vendor.md [hunk 13]
```diff
@@ -1,9 +1,100 @@
...
+Using Private Packagist for Vendors, your customers can install packages with Composer using a unique URL and an authentication token. You can create customers and manage their package access on the vendor page in your organization.
 
 Customers can use your packages in either an existing or a new Composer project.
 
+## Setting up a customer in Private Packagist
+
+You can manage your customers and grant them access to packages on the vendor page in Private Packagist.
+To create your first customer click on "Create Customer" and fill out the form on the next page.
+
+* Customer short name: the name will be used in the Composer repository URL and the web UI URL where your customer sees packages
```
Comment (naderman) [inline docs/setup-vendor.md:12]:
```suggestion
* Customer short name: The short name will be used in the Composer repository URL and the web UI URL where your customer sees packages
```

File: docs/setup-vendor.md [hunk 14]
```diff
@@ -1,9 +1,100 @@
...
+they have access to and can download archives. Please be aware that changing the short name will change the web UI URL and
+the Composer repository URL and will therefore break existing Composer projects referring to the current URL.
+* Minimum accesible stability: the setting allows you to limit which versions of a package a customer has access to. More about that in the
```
Comment (naderman) [inline docs/setup-vendor.md:15]:
```suggestion
* Minimum accesible stability: This setting allows you to limit which versions of a package a customer has access to. More about that in the
```

File: docs/setup-vendor.md [hunk 15]
```diff
@@ -1,9 +1,100 @@
...
+["Limit package access by minimum stability"](#limit-package-access-by-minimum-stability) section. 
+* Deliver source code URLs to customer through Composer: by default, Private Packagist removes source URLs for all packages
```
Comment (naderman) [inline docs/setup-vendor.md:17]:
```suggestion
* Deliver source code URLs to customers through Composer: By default, Private Packagist removes source URLs for all packages
```

File: docs/setup-vendor.md [hunk 16]
```diff
@@ -1,9 +1,100 @@
...
+assigned to a customer, e.g. for a git repository the URL to clone the repository, and customers are only able to install
+packages as archives. If your customer has access to your repositories and you want them to be able to install
+packages from source e.g. via git clone then you can enable the source code URL delivery to customers through Composer.
```
Comment (naderman) [inline docs/setup-vendor.md:20]:
```suggestion
packages from source, e.g. via git clone, then you can enable the source code URL delivery to customers through Composer.
```

**Review by naderman — CHANGES_REQUESTED** [review]

**Pull request merged**